pkg/controllers/securitypolicy_controller.go

```diff
@@ -27,82 +29,108 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/util"
 )
 
-var (
-	log = logf.Log.WithName("controller").WithName("securitypolicy")
-)
-
 const (
 	WCP_SYSTEM_RESOURCE = "vmware-system-shared-t1"
 	METRIC_RES_TYPE     = "securitypolicy"
 )
 
-// SecurityPolicyReconciler reconciles a SecurityPolicy object
+var (
+	log           = logf.Log.WithName("controller").WithName("securitypolicy")
+	resultNormal  = ctrl.Result{}
+	resultRequeue = ctrl.Result{Requeue: true}
+)
+
+// SecurityPolicyReconciler SecurityPolicyReconcile reconciles a SecurityPolicy object
 type SecurityPolicyReconciler struct {
 	Client  client.Client
-	Scheme  *runtime.Scheme
+	Scheme  *apimachineryruntime.Scheme
 	Service *services.SecurityPolicyService
 }
 
+func updateFail(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy, e *error) {
+	r.setSecurityPolicyReadyStatusFalse(c, o, e)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, METRIC_RES_TYPE)
+}
+
+func deleteFail(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy, e *error) {
+	r.setSecurityPolicyReadyStatusFalse(c, o, e)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteFailTotal, METRIC_RES_TYPE)
+}
+
+func updateSuccess(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy) {
+	r.setSecurityPolicyReadyStatusTrue(c, o)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, METRIC_RES_TYPE)
+}
+
+func deleteSuccess(r *SecurityPolicyReconciler, c *context.Context, o *v1alpha1.SecurityPolicy) {
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteSuccessTotal, METRIC_RES_TYPE)
+}
+
 func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	obj := &v1alpha1.SecurityPolicy{}
 	log.Info("reconciling securitypolicy CR", "securitypolicy", req.NamespacedName)
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, METRIC_RES_TYPE)
 
 	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		log.Error(err, "unable to fetch security policy CR", "req", req.NamespacedName)
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		log.V(1).Info("fetch securitypolicy CR", "req", req.NamespacedName)
```

Comment:
After you remove "unable to", from `V(1).Info` we know this is an error message?

Reply:
Since we ignore not found and take it as normal result, I think we could not pay specific attention to it? This error rarely happens, though.

Reply:
From your new log, can we know it's about not found?

Reply:
Added back.